### PR TITLE
Validate update split 1

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2158,22 +2174,22 @@ static bool verify_renderpass_compatibility(layer_data *my_data, const VkRenderP
...
-        errorStr << "VkPipelineLayout (" << layout << ") only contains " << pipeline_layout_it->second.descriptorSetLayouts.size()
-                 << " setLayouts corresponding to sets 0-" << pipeline_layout_it->second.descriptorSetLayouts.size() - 1
+        errorStr << "VkPipelineLayout (" << layout << ") only contains " << pipeline_layout->descriptorSetLayouts.size()
+                 << " setLayouts corresponding to sets 0-" << pipeline_layout->descriptorSetLayouts.size() - 1
                  << ", but you're attempting to bind set to index " << layoutIndex;
         errorMsg = errorStr.str();
         return false;
     }
-    auto layout_node = my_data->descriptorSetLayoutMap[pipeline_layout_it->second.descriptorSetLayouts[layoutIndex]];
+    auto layout_node = pipeline_layout->setLayouts[layoutIndex];
```
Comment (tobine) [inline layers/core_validation.cpp]: nice

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -3410,7 +3426,8 @@ static bool validateIdleDescriptorSet(const layer_data *my_data, VkDescriptorSet
 // Verify that given pool has descriptors that are being requested for allocation.
 // NOTE : Calls to this function should be wrapped in mutex
 static bool validate_descriptor_availability_in_pool(layer_data *dev_data, DESCRIPTOR_POOL_NODE *pPoolNode, uint32_t count,
-                                                     const VkDescriptorSetLayout *pSetLayouts) {
+                                                     std::vector<cvdescriptorset::DescriptorSetLayout const *> const & layout_nodes,
+                                                     uint32_t requiredDescriptorsByType[]) {
```
Comment (tobine) [inline layers/core_validation.cpp]: Perhaps update this function name and comment?
Now includes update to requiredDescriptorsByType

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -5902,61 +5899,58 @@ AllocateDescriptorSets(VkDevice device, const VkDescriptorSetAllocateInfo *pAllo
...
-                        return VK_ERROR_VALIDATION_FAILED_EXT;
-                    }
-                } else {
-                    // Insert set into this pool
+                if (layout_nodes[i]) {
+                    auto pNewNode = new cvdescriptorset::DescriptorSet(
+                            pDescriptorSets[i], layout_nodes[i], &dev_data->bufferMap, &dev_data->memObjMap, &dev_data->bufferViewMap,
+                            &dev_data->samplerMap, &dev_data->imageViewMap, &dev_data->imageMap,
+                            &dev_data->device_extensions.imageToSwapchainMap, &dev_data->device_extensions.swapchainMap);
+
```
Comment (tobine) [inline layers/core_validation.cpp]: Yeah, this is better

**Comment by tobine** [comment]
Just that one function that could probably use an updated name and comment
LGTM :+1:

**Comment by chrisforbes** [comment]
Rename done.

**Comment by chrisforbes** [comment]
Cherry-picked to master.

**Pull request merged**